pkg/wallet/wallet.go

```diff
@@ -7,50 +7,43 @@ import (
 	"crypto/rand"
 	"crypto/sha256"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"os"
 	"strings"
 
-	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/base58"
+	"github.com/btcsuite/btcutil/base58"
 	"golang.org/x/crypto/pbkdf2"
-	"gopkg.in/yaml.v3"
 	"lukechampine.com/blake3"
 )
 
-var ErrUnprotectedSerialization = errors.New("private key must be protected before serialization")
-
 const (
 	SecretKeyLength           = 32
 	PBKDF2NbRound             = 10000
 	FileModeUserReadWriteOnly = 0o600
-	MinAddressLength          = 49
+	Base58Version             = 0x00
 )
 
-//nolint:tagliatelle
+// KeyPair structure contains all the information necessary to save a key pair securely.
 type KeyPair struct {
-	PrivateKey []byte   `yaml:",flow"`
-	PublicKey  []byte   `yaml:",flow"`
-	Salt       [16]byte `yaml:",flow"`
-	Nonce      [12]byte `yaml:",flow"`
-	Protected  bool
+	PrivateKey []byte
+	PublicKey  []byte
+	Salt       [16]byte
+	Nonce      [12]byte
 }
 
-//nolint:tagliatelle
+// Wallet structure allows to link a nickname, an address and a version to one or more key pairs.
 type Wallet struct {
-	Version  uint8     `json:"version"`
-	Nickname string    `json:"nickname"`
-	Address  string    `json:"address"`
-	KeyPairs []KeyPair `json:"keyPairs"`
-}
-
-type Config struct {
-	// address
-	Wallets []KeyPair `json:"wallets"`
+	Version  uint8
+	Nickname string
+	Address  string
+	KeyPair  KeyPair
 }
 
-func (w *Wallet) Protect(password string, keyPairIndex uint8) error {
```

Comment:
Protect and unProtect function contains duplicated code that could be factorized